### ISSUE TITLE
Update/Refresh behaviour

### ISSUE THREAD

**Issue by Joniator** [FreshRSS/FreshRSS#2262]
Im wondering how exactly does refreshing feeds behave?
It feels like refreshing doesn't really do something most of the time.
It just immediately hits you with "No feeds to refresh", until at somepoint it does not and updates every feed with many articles.
But this seems related to the "Do not automatically refresh more often than", which might be better named "Do not refresh more often than" then, since it doesn't seem to matter if automatic or not.

I think that one of the following options should be more clear to the user:

- Either rename the setting (Name seems consistently misleading at least in english and german) and make the possible values more based on reality. A few cheap requests like this don't hurt if you fire them up every 1-5 Minutes, and 20 minutes minimum seems overly defensive to me.
- Make sure this only affects automatic refreshs, not the one issued by the user via the interface.

If you can rely on the "Do not automatically refresh more often than" setting, it would be easily possible to run the cron script every minute, and check if the time is reached. This would need updated install instructions, where the cron script is run 'hourly', which seems like an eternity for me.

FreshRSS-Version: 1.13.1
PHP-Version: 7.2.14

**Comment by Frenzie** [comment]
> If you can rely on the "Do not automatically refresh more often than" setting, it would be easily possible to run the cron script every minute, and check if the time is reached.

That's how it works, yes.

> This would need updated install instructions, where the cron script is run 'hourly', which seems like an eternity for me.

That's an example, not an instruction. I use 15 minutes in cron, combined with a 30 minute default in FreshRSS, as well as a few custom per-feed settings. That's mainly higher, .e.g, I've got a comic set to once a week so it shows up in the weekend. (I would support a PR to change the example to 15 minutes.)

Many feeds have WebSub, btw.

**Comment by Alkarex** [comment]
Yes indeed, we should update the examples and terminology.
The examples we give for the Docker deployment are faster:
https://github.com/FreshRSS/FreshRSS/tree/master/Docker#cron-job-to-automatically-refresh-feeds

**Comment by Joniator** [comment]
> That's how it works, yes.

Okay, at I'm not doing anything wrong then.
But the point still stands, for me at least 20 Minutes until update seems limiting for no real reason.
I like to see updates  as soon as possible, and maybe someone watches a gitlab teamstream and wants to see events rather soon than later, in which case 20 Minutes feel like an eternity

Another point still stands: It doesn't update if you press the update button if the time isn't reached.
This is great behaviour for your comic strip, but seems counter-intuitive for news, especially since you are still bound to the 20 Minute delay in the best worst case.

So for my main feature request: 
- Give more options for the time (1min, 5min, 10min)

I'm looking into it, and as far as I can tell it should almost be enough to just change the view and add the second/text mapping, or am I missing something?

Changing the example/bringing it on par with the docker one would be a nice change too, since less technical users might want to go just copy&paste everything and then wondering why it does auto updates so infrequently.

**Comment by Frenzie** [comment]
> Give more options for the time (1min, 5min, 10min)

Oh, now I understand! I didn't get that from the OP. I thought you were basically just slightly dissatisfied about the defaults and/or the examples. Curiously, the per-feed config does allow for 15 mins.

It's just a matter of what the UI shows. If you right-click inspect element and change the value from 1200 to, say, 200 and save, that'll be the default until you change it. Note that if you refresh too often (more often than every 10-15 minutes) then some servers might ban you. Too easily exposing <5 minutes might not be a good idea.

**Comment by Joniator** [comment]
Right-clicking is always a dirty hack, and IMO an application shouldn't push too much opinion on its users. If they want to risk bans, warn them, but let them.

**Comment by Alkarex** [comment]
> If they want to risk bans, warn them, but let them.

@Joniator No, sorry. We are part of an ecosystem. A typical reaction that we have seen from several servers is to simply ban by user-agent (that is why we had to remove "SimplePie" from our user-agent string, because it was too frequently banned), or to remove their RSS feed altogether. Bad user behaviours affect the larger community.

If you need (near-)realtime updates, use the WebSub push mechanism, which is exactly meant for that, and is very easy to deploy server-side. FreshRSS has built-in client support for it.

**Comment by Joniator** [comment]
Alright, I'm rather new using rss and wasn't aware that the perfomance implications would be that big of a deal, but I'll refrain from setting the numbers down then.

Regarding WebSub: Is it assumed that a rss feed supports it by linking a hub in the metadata?
I haven't seen it mentioned in any of the feeds I subscribe to, and resources on it are rather simple, especially in the context of rss.

Edit:
Nevermind, I just saw that some few feeds have this setting enabled, while others dont, so I guess that it is hit or miss, but if the feed supports it the metadata is in the feed.

**Comment by squromiv** [comment]
>  "Do not automatically refresh more often than", which might be better named "Do not refresh more often than"

+1. It confuses me too. Also I think, that this setting should be replaced to other menu category. For example "General" or "Misc". Because it has no relation to "Archiving/Purge", but  presupposes fewer requests to news sites.
>No, sorry. We are part of an ecosystem.

It is reasonable remark, but I also ran into a problem, when I need to have updates for forum topic practically in real time. Whatever it sounds like, but I miss the feature of lower time values too.

**Comment by math-GH** [comment]
How is the status? It had been done a lot since V1.13

**Comment by Alkarex** [comment]
What remains is to re-organise a bit some of the options, and I guess we have other tickets to track that

**Comment by squromiv** [comment]
> What remains

What was done regarding this issue? Did I miss anything?
"Do not automatically refresh more often than" option is not renamed and still is placed in "Archiving" submenu. Refresh time low limit can not be less than 20 (15) minutes.

**Comment by Alkarex** [comment]
This issue was a mix of different things. We will not allow a lower limit for now (or maybe only for local networks), due to well-known abuses that are risky for the ecosystem. Faster refresh scenarios must implement another protocol such as WebSub. I believe we already have another ticket about reorganising the options.

**Comment by math-GH** [comment]
A little bit similar with the FR around the feed errors: https://github.com/FreshRSS/FreshRSS/labels/FR%3A%20handle%20error%20feeds

**Comment by limdingwen** [comment]
Just wanted to thank you for adding this feature. Back when I was on Miniflux (my use case was subscribing to hundreds of feeds from the same server), I instantly hit into DDoS protection when I imported my feeds. I was lucky I was blocked by CloudFlare (which was temporary) instead of the server itself.

Wanted to create an extension on FreshRSS to "stagger" the feed refreshes, and apparently it's built-in! Another thing that helps is that feed refreshes are done sequentially (according to my RSSHub logs), instead of doing it all at once.

**Comment by nhan000** [comment]
Hi, what's the meaning of the Mute checkbox here?
![image](https://github.com/FreshRSS/FreshRSS/assets/85216095/2c85cabc-84c5-47cc-8248-c9f641ccaa73)

**Comment by Alkarex** [comment]
> Hi, what's the meaning of the Mute checkbox here?

See https://github.com/FreshRSS/FreshRSS/discussions/5581
